synapse/federation/sender/per_destination_queue.py

```diff
@@ -138,7 +155,11 @@ def send_pdu(self, pdu: EventBase) -> None:
         Args:
             pdu: pdu to send
         """
-        self._pending_pdus.append(pdu)
+        if not self._catching_up:
```

Comment:
if this is hit while `_catching_up` is None, then the PDU will get dropped?

Reply:
Not the case, but agree the readability is horrible like this -- should be sorted now :)